src/custom/state/orders/updater.ts

```diff
@@ -128,6 +133,73 @@ const constructGetLogsRetry = (provider: Web3Provider) => {
   return getLogsRetry
 }
 
+export function EventUpdaterApiOnly(): null {
+  const { chainId, library } = useActiveWeb3React()
+
+  const pending = usePendingOrders({ chainId })
+  const fulfillOrdersBatch = useFulfillOrdersBatch()
+
+  useEffect(() => {
+    let interval: NodeJS.Timeout | null = null
+    console.log('EventUpdaterApiOnly::useEffect')
+
+    if (!chainId || pending.length <= 0) {
+      console.log('EventUpdaterApiOnly::exiting', chainId, pending.length)
+
+      return
+    }
+
+    async function update(): Promise<void> {
```

Comment:
So all the updater function feels a bit big. is it possible to move all the logic the update logic to a function? 
So lines 152-191 would disappear, and we would add as parameter any dependency:
i.e. `update({ lastBlockNumber, pending, fulfillOrdersBatch })`